File: app/src/main/java/com/quran/labs/androidquran/di/component/activity/QuranActivityComponent.java
package com.quran.labs.androidquran.di.component.activity;

import com.quran.labs.androidquran.di.ActivityScope;
import com.quran.labs.androidquran.di.module.activity.QuranActivityModule;
import com.quran.labs.androidquran.ui.QuranActivity;
import dagger.Subcomponent;

@ActivityScope
@Subcomponent(modules = QuranActivityModule.class)
public interface QuranActivityComponent {
  void inject(QuranActivity quranActivity);

  @Subcomponent.Builder
  interface Builder {
    QuranActivityComponent build();
  }
}

File: app/src/main/java/com/quran/labs/androidquran/di/component/activity/QuranActivityComponent.kt
package com.quran.labs.androidquran.di.component.activity

import com.quran.data.di.QuranActivityLevelScope
import com.quran.data.di.QuranActivityScope
import com.quran.labs.androidquran.di.module.activity.QuranActivityBindingContainer
import com.quran.labs.androidquran.ui.QuranActivity
import dev.zacsweers.metro.GraphExtension

@QuranActivityScope
@GraphExtension(scope = QuranActivityLevelScope::class, bindingContainers = [QuranActivityBindingContainer::class])
interface QuranActivityComponent {
  fun inject(quranActivity: QuranActivity)

  @GraphExtension.Factory
  interface Factory {
    fun generate(): QuranActivityComponent
  }
}
